sstables/sstables.cc

```diff
@@ -2412,7 +2412,8 @@ future<> sstable::write_components(
     }
     return seastar::async([this, mr = std::move(mr), estimated_partitions, schema = std::move(schema), cfg, stats, &pc] () mutable {
         auto wr = get_writer(*schema, estimated_partitions, cfg, stats, pc);
-        mr.consume_in_thread(std::move(wr), db::no_timeout);
```

Comment:
I think the most important part of the change is the fact that the sstable writer now validates the input mutation fragment stream. This cannot be deduced from the commit message ("flat_mutation_reader: mutation_fragment_stream_validator"). I would suggest "sstables: writer: Validate input mutation fragment stream". The addition of the validator could be a separate patch, but I am not going to insist on that.

Reply:
sure, makes sense.